upload/catalog/controller/extension/smailyforopencart/rss.php

```diff
@@ -8,16 +8,27 @@ public function index() {
         $this->load->model('tool/image');
         $this->load->model('localisation/currency');
 
-        // Get params from url for filtering products.
+        $category = null;
+        $limit = 50;
+        $sort = 'p.date_added';
+        $order = 'DESC';
+        // Get params from URL for filtering products.
         if (isset($this->request->get['category'])) {
             $category = $this->request->get['category'];
-        } else {
-            $category = null;
+            // Category parameter can be either in string format or category ID.
+            $category = $this->getCategoryIdByName($category) !== '' ? $this->getCategoryIdByName($category) : (int) $category;
```

Comment:
Flip this condition around to avoid querying the database twice:

```suggestion
            $category = (int) $category > 0 ? (int) $category : $this->getCategoryIdByName($category);
```